### PR TITLE
New ALS DataGen and Benchmark implementation

### PR DESCRIPTION
**Pull request by xwu99**
The original implementation incorrect use "sparsity" #643, and generate same data for both direct and implicit ratings and runs slowly. The new ALS benchmark implements the following changes: 

* A new RatingDataGenerator following MovieLens rating rules for explicit and implicit ratings and better performance
* Use rating number directy rather than use sparsity
* A new benchmark code to use new ml.ALS dataframe-based APIs
* Revise related configs according to the above changes

### CODE REVIEW

**Comment by xwu99** [comment]
@minmingzhu @carsonwang

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/RatingDataGenerator.scala [hunk 1]
```diff
@@ -46,35 +46,45 @@ object RatingDataGenerator {
...
-    val rng = new java.util.Random()
-    val data = rawData.map{v =>
-      val a = Array.fill[Double](v.size)(0.0)
-      v.foreachActive{(i,vi) =>
-         if(rng.nextDouble <= sparsity){
-           a(i) = vi
-         }
+    val ratingData = sc.parallelize(1 to numRatings, numPartitions)
+      .mapPartitions { p =>
+        val rng = new java.util.Random()
```
Comment (carsonwang) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/RatingDataGenerator.scala:70]: Let's use `mapPartitionsWithIndex` and use the partition index as the seed. This way, we can generate same data for different runs and each partition will use different seeds. 
```
import scala.util.hashing
val rng = new java.util.Random((hashing.byteswap32(index))
```

**Comment by carsonwang** [comment]
could you please show some sample data generated? Let's make sure each partition has different data, and if we generate the data twice using same configuration they are identical.

**Comment by xwu99** [comment]
> could you please show some sample data generated? Let's make sure each partition has different data, and if we generate the data twice using same configuration they are identical.

After fixed seed with partition index, they are the same from run to run. some data:

```
+----+----+------+
|user|item|rating|
+----+----+------+
|  60|  48|   2.5|
|  47|  15|   1.5|
|  91|  61|   2.5|
|  54|  77|   0.5|
|  73|  62|  -1.5|
|  44|  84|  -1.5|
|  41|  20|   1.5|
|  88|  24|   0.5|
|  52|  60|   1.5|
|  82|  92|   1.5|
|  45|  45|   0.5|
|  87|   2|   0.5|
|  25|  53|   1.5|
|  35|  60|  -1.5|
|  55|  30|   1.5|
|  91|  74|  -0.5|
|  12|  62|   2.5|
|  77|  16|  -0.5|
|   7|  16|   1.5|
|  37|  43|   0.5|
+----+----+------+
```

**Comment by carsonwang** [comment]
Thanks. Merged.

**Pull request merged**